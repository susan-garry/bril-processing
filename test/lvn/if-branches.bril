# ARGS: true
# CMD: bril2json < {filename} | python3 ../../lvn.py | python3 ../../dce_trivial.py | bril2txt

@main(input:bool) {
  x: int = const 0;
  br input .ift .iff;
.ift:
  x: int = const 1;
  jmp .end;
.iff:
  x: int = const 2;
.end:
  print x;
}