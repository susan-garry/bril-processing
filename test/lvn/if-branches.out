@main(input: bool) {
  x: int = const 0;
  br input .ift .iff;
.ift:
  x: int = const 1;
  jmp .end;
.iff:
  x: int = const 2;
.end:
  print x;
}
